src/de/dkfz/b080/co/common/WorkflowUsingMergedBams.groovy

```diff
@@ -1,144 +1,173 @@
-                BasicBamFile[] copy = new BasicBamFile[found.length];
+                BasicBamFile[] copy = new BasicBamFile[found.length]
                 for (int i = 0; i < found.length; i++) {
-                    if (found[i] == null) continue;
-                    copy[i] = new BasicBamFile(new BaseFile.ConstructionHelperForSourceFiles(found[i].getPath(), context, found[i].getFileStage().copy(), null));
-                    copy[i].setAsSourceFile();
+                    if (found[i] == null) continue
+                    copy[i] = new BasicBamFile(new BaseFile.ConstructionHelperForSourceFiles(found[i].getPath(), context, found[i].getFileStage().copy(), null))
+                    copy[i].setAsSourceFile()
                 }
-                found = copy;
+                found = copy
             }
         }
-        return found;
+        return found
     }
 
-    public boolean checkInitialFiles(ExecutionContext context, BasicBamFile[] initialBamFiles) {
-        if (!initialBamFiles) initialBamFiles = new BasicBamFile[2];
-        if (initialBamFiles.size() == 1) initialBamFiles = [initialBamFiles[0], null] as BasicBamFile[];
-        BasicBamFile bamControlMerged = initialBamFiles[0];
-        BasicBamFile bamTumorMerged = initialBamFiles[1];
+    boolean checkInitialFiles(ExecutionContext context, BasicBamFile[] initialBamFiles) {
         boolean isNoControlWorkflow = getflag(context, IS_NO_CONTROL_WORKFLOW, false)
-        if ((!isNoControlWorkflow && bamControlMerged == null) || bamTumorMerged == null) {
-            if (bamControlMerged == null && !isNoControlWorkflow)
-                context.addErrorEntry(ExecutionContextError.EXECUTION_NOINPUTDATA.expand("Control bam is missing and workflow is not set to accept tumor only"));
-            if (bamTumorMerged == null)
-                context.addErrorEntry(ExecutionContextError.EXECUTION_NOINPUTDATA.expand("Tumor bam is missing"));
-            return false;
+        if (!initialBamFiles) {
+            context.addErrorEntry(ExecutionContextError.EXECUTION_NOINPUTDATA.expand("Did not find any bam files."))
+            return false
+        }
+        if (isNoControlWorkflow) {
+            boolean foundAll = true
+            initialBamFiles.each { BasicBamFile it -> foundAll &= it && ((COFileStageSettings) it.getFileStage()).sample.sampleType == Sample.SampleType.TUMOR }
+
+            if (!foundAll) {
+                context.addErrorEntry(ExecutionContextError.EXECUTION_NOINPUTDATA.expand("Not all found files are tumor bam files."))
+                return false
+            }
+
+        } else {
+            boolean foundAll = true
+            BasicBamFile bamControlMerged = initialBamFiles[0]
+
+            for (int i = 1; i < initialBamFiles.size(); i++) {
+                if (initialBamFiles[i] == null || !((COFileStageSettings) initialBamFiles[i].getFileStage()).sample.sampleType == Sample.SampleType.TUMOR) {
```

Comment:
Maybe do `initialBamFiles.collect { it != null }.all` at the beginning, to get rid of the if's down here.